### ISSUE TITLE
Long test execution times after Eclipse 2023-06 update

### ISSUE THREAD

**Issue by ptziegler** [eclipse-windowbuilder/windowbuilder#481]
Our test runs frequently take ~1h to complete, rather than the "normal" 35min. Some even hit the timeout after 1h30min. See [here](https://github.com/eclipse-windowbuilder/windowbuilder/actions/runs/5283488903/jobs/9559799595) as an example.

In the past, he occasionally had this problem while testing against the previous (2022-09) version, but I have the impression this problem got worse after the update.

Given that the jobs usually run slower on Windows than on Linux, that's where we should start looking.

**Comment by ptziegler** [comment]
Looking at [this](https://github.com/eclipse-windowbuilder/windowbuilder/actions/runs/5293487420/jobs/9581676877) job, the entire build took 1h6min, but only 33min were spent on running the test.

Meaning an additional 33min was spent just on loading the target platform and compiling the code. Running the job locally, I'm pretty sure the release update site is the main problem. I think that also explains why those builds sporadically run into a timeout.

```
Warning:  Some attempts to read artifact osgi.bundle,org.eclipse.core.resources,3.19.0.v20230510-1240 failed:
Warning:     An error occurred while transferring artifact canonical: osgi.bundle,org.eclipse.core.resources,3.19.0.v20230510-1240 from repository https://download.eclipse.org/releases/2023-06/202306141000:
Warning:        Retry another mirror:
Warning:           download from https://espejito.fder.edu.uy/eclipse/releases/2023-06/202306141000/plugins/org.eclipse.core.resources_3.19.0.v20230510-1240.jar failed
```

**Comment by ptziegler** [comment]
And the fallback mirror is painfully slow... Which is why the download only takes forever.

**Comment by merks** [comment]
I've had this happen on the BIRT builds where one mirror it was using was extremely slow and it used that mirror to get the JustJ JRE such that the one file transfer took almost 1/2 hour.  The next day the problem didn't happen anymore.

**Comment by ptziegler** [comment]
Hm... I also got the same, slow resolution time after I went back to the 2022-09 target-platform definition, so this really seems to be the case.

Despite that, if I run the builds in short succession, once with Tycho 3 and once with Tycho 4, the latter is significantly faster. The last Jenkins build only took [26min](https://ci.eclipse.org/windowbuilder/job/build/job/PR-483/2/), so I assume they made several performance improvements since the previous release.

**Comment by ptziegler** [comment]
I think this is resolved with #483. Whenever I noticed those long target-platform resolution times, I switched over to the Tycho 4 branch and immediately noticed major performance improvements. In the end, I think its an accumulation of improvements that make the upcoming version much more efficient.

I also think the current nightly builds are stable enough that we can use them for the nightly builds.
